### PR TITLE
Add signatures for extraction functions.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Extracted from https://github.com/Homebrew/brew/pull/14735.

### CODE REVIEW

File: Library/Homebrew/unpack_strategy.rb [hunk 1]
```diff
@@ -135,14 +135,27 @@ def initialize(path, ref_type: nil, ref: nil, merge_xattrs: nil)
   def extract_to_dir(unpack_dir, basename:, verbose:); end
   private :extract_to_dir
 
-  def extract(to: nil, basename: nil, verbose: nil)
+  sig {
+    params(
+      to: T.nilable(Pathname), basename: T.nilable(T.any(String, Pathname)), verbose: T::Boolean,
```
Comment (dduugg) [inline Library/Homebrew/unpack_strategy.rb:140]: Not necessarily for this PR, but i wonder if want to create a type alias for this one:
```
$ git grep -o 'T.any(String, Pathname)' | wc -l
56
```

Comment (MikeMcQuaid) [reply Library/Homebrew/unpack_strategy.rb:140]: @dduugg Good idea!

File: Library/Homebrew/download_strategy.rb [hunk 2]
```diff
@@ -95,10 +95,10 @@ def quiet?
   # @api public
   def stage(&block)
     UnpackStrategy.detect(cached_location,
-                          prioritise_extension: true,
+                          prioritize_extension: true,
```
Comment (MikeMcQuaid) [inline Library/Homebrew/download_strategy.rb:98]: As mentioned elsewhere: let's not rename this or, if we want to discuss renaming, make a PR that only renames and does nothing else.

Comment (reitermarkus) [reply Library/Homebrew/download_strategy.rb:98]: https://github.com/Homebrew/brew/pull/14735#discussion_r1116446511

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks again @reitermarkus!

**Pull request merged**